### DISCUSSION TITLE
Prevent renaming libraries

### DISCUSSION THREAD

**Discussion by peastman** [pypa/auditwheel#511] [Q&A]
When I use auditwheel to copy libraries into my wheel, it renames all of them, adding unique hashes to them.  This breaks my wheel: the names of the libraries are part of the public interface.  They're being distributed with the expectation that other programs will link against them.  How can I prevent that from happening?

As an example of what I want to do, if I `pip install torch` and then look inside `site-packages/torch/lib`, the libraries have ordinary names like `libtorch.so` and `libc10.so`, not `libtorch-12345678.so`.  I want to do the same thing with my package.

**Comment by mayeut** [comment]
You need to tell your build backend to include these libraries in the original wheel.
AuditWheel will not rename libraries already present the original wheel. Only libraries that are grafted from outside the original wheel get renamed (you can see this as: they're external dependencies that should not conflict with other wheels so rename them).

**Comment by peastman** [comment]
Thanks!  That's what I was missing.
